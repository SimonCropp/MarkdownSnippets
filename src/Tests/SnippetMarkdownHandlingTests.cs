using System.Collections.Generic;
using System.IO;
using System.Text;
using ApprovalTests;
using MarkdownSnippets;
using Xunit;
using Xunit.Abstractions;

public class SnippetMarkdownHandlingTests : 
    TestBase
{
    [Fact]
    public void AppendGroup()
    {
        var builder = new StringBuilder();
        var snippets = new List<Snippet> {Snippet.Build(1, 2, "theValue", "thekey", "thelanguage", "thePath")};
        var gitHubSnippetMarkdownHandling = new SnippetMarkdownHandling(GitRepoDirectoryFinder.FindForFilePath());
        using (var writer = new StringWriter(builder))
        {
            gitHubSnippetMarkdownHandling.AppendGroup("key1", snippets, writer);
        }

        Approvals.Verify(builder.ToString());
    }

    public SnippetMarkdownHandlingTests(ITestOutputHelper output) :
        base(output)
    {
    }
}