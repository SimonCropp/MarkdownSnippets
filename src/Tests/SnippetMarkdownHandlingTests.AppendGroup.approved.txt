```thelanguage
theValue
```
<sup>[snippet source](thePath#L1-L2)</sup>
